### PR TITLE
Add erf_inv/1 to Nx and enable in EXLA

### PR DESCRIPTION
**Pull request by elbow-jason**
This PR adds `erf_inv/1` operation to `Nx`, `Nx.Backend`, and `EXLA`.

 #### SciPy:
 
 ```python
 In [2]: from scipy import special

In [3]: special.erfinv(0.999)
Out[3]: 2.3267537655135464

In [4]: special.erfinv(0.01)
Out[4]: 0.008862501280950608
 ```
 
 #### Nx:
 ```elixir
 iex(1)> Nx.erf_inv(0.999)
#Nx.Tensor<
  f64
  2.326753756865462
>
iex(2)> Nx.erf_inv(0.01) 
#Nx.Tensor<
  f64
  0.008862500728738846
>
 ```
 
 #### Resouces:

  + [erf_inv in XLA](https://github.com/tensorflow/tensorflow/blob/0e6b3a307f6363a9ee9dc9e03a8a0350c7668799/tensorflow/compiler/xla/client/lib/math.cc#L340-L390)
  
  + [erf_inv's grad in Jax](https://github.com/google/jax/blob/7e040c0a89b7b1f5c3607623d67c811d977470f4/jax/_src/lax/lax.py#L2434-L2436)

### CODE REVIEW

File: nx/lib/nx/shared.ex [hunk 1]
```diff
@@ -249,6 +250,56 @@ defmodule Nx.Shared do
...
+    #Nx.Tensor<
+      f64
+      2.326753756865462
+    >
+
+    iex> Nx.erf_inv(0.01)
+    #Nx.Tensor<
+      f64
+      0.008862500728738846
+    >
```
Comment (josevalim) [inline nx/lib/nx/shared.ex:268]: Please indent with four spaces!

Comment (josevalim) [reply nx/lib/nx/shared.ex:268]: Also, we shouldn't be testing the `Nx` API via `Nx.Shared` doctests. We should either test the Nx.Shared API directly or add the test to `nx/shared_test`. :)

**Comment by josevalim** [comment]
Looks great, just one minor comment. :)

File: nx/lib/nx/shared.ex (hunk 1 above)
Comment (elbow-jason) [reply nx/lib/nx/shared.ex:268]: Good point. I missed that.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**